### DISCUSSION TITLE
Add an easy way to modify the values set in [Export] variables from script

### DISCUSSION THREAD

**Discussion by valkyrienyanko** [godotengine/godot-proposals#5318] [Scripting]
I have the variable `[Export] public string[] Entities;`. I set the values from the Godot editor. For this particular scenario, they are enemies I'm using to activate when a player enters a specific Area2D. Sometimes while I'm designing the level I'll remove the enemy from the level but I will also have to manually remove it from this [Export] variable I have set. Even though there is a `GetNodeOrNull()` function being used so there are no errors being thrown, it would be nice to have a way in the script to search for which nodes are null when pressing a button from the editor inspector or when the game runs. This way no unneeded checks are done when the actual game is shipped for people to play just because I forgot to update some [Export] values. This is not a big deal and I can live without the feature, it would just be a nice to have.

**Comment by theraot** [comment]
In your example, you have an array of `String`, I don't expect Godot to manipulate `String`s based on `Node`s because it won't consider the `String`s as references to `Node`s.

*And if you were to implement an addon or even a core feature to search for these, how would you tell apart innocent `String`s that aren't meant to refer to a `Node` from `String`s that are meant to refer to a `Node` that is not there?*

Instead, you should use `NodePath`s. *It has been a good while since I used C# with Godot, and I remember exports were an issue, but I also know they have improved since…* So, try exporting `NodePath`s instead. Godot should be able to keep track of those.

---

Now if that does not work, I suppose you would need a custom addon, one that you can configure to only check the properties you want it to (It could search the scenes for `Node`s of your custom type).

But before you go on making a custom addon. Consider some alternatives:

First of all, good all `prints`. In your editor you would see them in the output panel, and - by default on PC - they will go to a log. If you print every time one of those checks fail, that will help you identify the ones you need to remove during testing.

Similarly, consider making the script of your custom `Node` a tool script, then it can check and report if there is any problem in `_get_configuration_warning` (`_get_configuration_warnings` in Godot 4).

---

Anyway, all these approaches do not cover the case where you forgot to include a particular enemy. Thus, how about this: make a `tool` script (it can be the script of the custom `Node` itself) that generates the list. It would not only make sure the list is correct, but it will also save you time in populating it.

---

*I'll remind you that you can check `Engine.editor_hint` to know if your script is running in the editor, you might also be interested in `OS.is_debug_build`.*
